app/src/main/java/io/github/gsantner/memetastic/activity/MainActivity.java

```diff
@@ -687,4 +732,24 @@ public boolean onOptionsItemSelected(MenuItem item) {
         return handleBarClick(item);
     }
 
+    @Override
+    public void onPageScrolled(int i, float v, int i1) {
+
+    }
+
+    @Override
+    public void onPageSelected(int i) {
+        app.settings.setLastSelectedTab(i);
+    }
+
+    @Override
+    public void onPageScrollStateChanged(int i) {
+
+    }
+
```

Comment:
isn't this only doing selectCreateTab?

Reply:
Only added the listener to fix the bug i told you about with tab navigation not returning you to the last selected tab. Before the viewpager the last tab selected was updated in the onTabSelected method

Reply:
I didnt mark the listener things, but the method on the very bottom

Reply:
The method is being called from the adapter to navigate the user away from the hidden tab when the last item is removed